### PR TITLE
feat: Lemon Dialogs

### PR DESCRIPTION
**Pull request by benjackwhite**
## Problem

[Related Figma area](https://www.figma.com/file/Y9G24U4r04nEjIDGIEGuKI/PostHog-Design-System-One?node-id=3139%3A1388)

## Changes

* Adds a LemonDialog component to replace AntD's `Modal.confirm()` functionality
Screenshot 2022-08-10 at 16 44 24
Screenshot 2022-08-10 at 16 44 34

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

* Storybook

### CODE REVIEW

**Review by clarkus — APPROVED** [review]

File: frontend/src/lib/components/LemonDialog/LemonDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    onClose,
+    primaryButton,
+    tertiaryButton,
+    secondaryButton,
+    ...props
+}: LemonDialogProps): JSX.Element {
+    const [isOpen, setIsOpen] = useState(true)
+
+    primaryButton = primaryButton || {
+        children: 'Okay',
```
Comment (Twixes) [inline frontend/src/lib/components/LemonDialog/LemonDialog.tsx:29]: I'm a bit uncomfortable with how generic the default "Okay" is. 😅 We should be specific with affirmative buttons in most cases.

**Review by pauldambra — APPROVED** [review]
LGTM

The `x` close button doesn't pass accessibility check in storybook because it has no `aria-label="Close"` but that seems to come from the `react-modal` dependency

**Review by Twixes — APPROVED** [review]
Not sure about the default primary button, but looks good overall

File: frontend/src/lib/components/LemonDialog/LemonDialog.tsx (hunk 1 above)
Comment (benjackwhite) [reply frontend/src/lib/components/LemonDialog/LemonDialog.tsx:29]: I'm pretty certain it will never get used but it's a fair point 👍

**Comment by benjackwhite** [comment]
@pauldambra we are in control of that button so I added the aria-label

File: frontend/src/lib/components/LemonModal/LemonModal.tsx [hunk 2]
```diff
@@ -55,7 +55,7 @@ export function LemonModal({
     const modalContent = (
         <>
             <div className="LemonModal__closebutton">
-                <LemonButton icon={<IconClose />} size="small" status="stealth" onClick={onClose} />
+                <LemonButton icon={<IconClose />} size="small" status="stealth" onClick={onClose} aria-label="close" />
```
Comment (pauldambra) [inline frontend/src/lib/components/LemonModal/LemonModal.tsx:58]: Ah! 🙌 

celebrate

**Review by pauldambra — APPROVED** [review]

File: frontend/src/lib/components/LemonModal/LemonModal.tsx (hunk 2 above)
Comment (benjackwhite) [reply frontend/src/lib/components/LemonModal/LemonModal.tsx:58]: You're the legend for noticing (and building the tooling to know that) this is missing!

**Pull request merged**